extras/plot_throughput/plot_data.py

```diff
@@ -0,0 +1,74 @@
+#!/usr/bin/env python3
+
+import pandas as pd
+import glob
+import subprocess
+import os
+
+# Create some useful directories
+try:
+    os.mkdir('clean')
+except OSError as error:
+    print(error)
+try:
+    os.mkdir('plots')
+except OSError as error:
+    print(error)
+
+# Get all csv input filenames
+filenames = glob.glob('*.csv')
+print("filenames = {}".format(" ".join(filenames)) )
+
+output_peaks = 'clean/peaks.csv'
+
+peaks = pd.DataFrame()
+
+for input_csv in filenames :
+    # Ignore the peaks file
+    if input_csv == output_peaks :
+        continue
+
+    # Create output filename in clean dir
+    output_csv = 'clean/{}'.format(input_csv)
+
+    # Read raw input data
+    raw = pd.read_csv(input_csv)
+    print("\nRead ", len(raw.index), " datapoints from ", input_csv)
+
+    # Get int value from input directory name
+    raw['mu'] = raw['directory'].str.extract('(\d+)').astype(int)
+
+    # Calculate throughput
+    raw['throughput (events/s)'] = raw.apply(lambda x: x['processed_events'] * 1e9 / x['processing_time'] , axis=1)
+    raw = raw[['mu', 'threads', 'throughput (events/s)']]
+
+    # Calculate mean throughput and respective standard deviation for each mu/thread pair
+    clean = raw.groupby(['mu', 'threads']).mean().fillna(0.0)
+    clean['stddev (events/s)'] = raw.groupby(['mu', 'threads']).std().fillna(0.0)['throughput (events/s)']
+
+    clean = clean.reset_index('mu')
+
+    # Separate each mu value to different columns
+    groups = clean.groupby('mu')
+
+    final = pd.DataFrame()
+    for x in groups.groups:
+        final['throughput mu{} (events/s)'.format(x)] = groups.get_group(x)['throughput (events/s)']
+        final['stddev mu{} (events/s)'.format(x)] = groups.get_group(x)['stddev (events/s)']
+
+    # Print to output csv file
+    final.to_csv(output_csv)
+    print("Printed", input_csv, " output to ", output_csv)
+
+    # Add maximum values to global comparison table
+    idx = clean.groupby('mu')['throughput (events/s)'].transform(max) == clean['throughput (events/s)']

```

Comment:
I went for something a bit different here because I wanted the stddev as well, so I need to keep the index as well.
```python
peak = clean.loc[
            clean["throughput (events/s)"]["mean"].groupby("mu").idxmax()
        ].reset_index("threads")
```